internal/objectstore/dynamic_cache.go

```diff
@@ -278,9 +279,22 @@ func (dc *DynamicCache) listFromInformer(ctx context.Context, key store.Key) (*u
 		l = informer.Lister().ByNamespace(key.Namespace)
 	}
 
+	if key.Selector != nil && key.RequirementsSelector != nil {
+		return nil, false, fmt.Errorf("must provide only one of Key.Selector and Key.RequirementsSelector")
+	}
+
 	var selector = kLabels.Everything()
 	if key.Selector != nil {
 		selector = key.Selector.AsSelector()
+	} else if key.RequirementsSelector != nil {
+		selector = kLabels.NewSelector()
+		for _, r := range *key.RequirementsSelector {
+			requirement, err := kLabels.NewRequirement(r.Key, r.Operator, r.StrValues)
+			if err != nil {
+				return nil, false, errors.Wrap(err, "unable to convert requirement to k8s requirement")
```

Comment:
no need to use a library

```suggestion
				return nil, false, fmt.Errorf("unable to convert requirement to k8s requirement: %w", err)
```